sunkit_image/utils/tests/test_noise.py

```diff
@@ -1,36 +1,38 @@
 import numpy as np
 from skimage import data
 
-from ..noise import NoiseLevelEstimation as nle
+import sunkit_image.utils.noise as nf
 
 img = data.camera()
 
 
-def convmtx2_test():
+def convmtx2_test(img):
 
-    tt = nle.convmtx2(img, 11.0, 7.0)
+    tt = nf.convmtx2(img, 11.0, 7.0)
     assert tt.shape == ((11 - img.shape[0] + 1) * (7 - img.shape[1] + 1), 11.0 * 7.0)
 
 
-def noiselevel_test():
+def noiselevel_test(img):
 
     noise_levels = np.array([5.0, 10.0, 20.0, 42.0])
     n_levels = np.zeros_like(noise_levels)
     n_patches = np.zeros_like(noise_levels)
 
     for n in range(noise_levels.size):
         noise = img + np.random.standard_normal(img.shape) * noise_levels[n]
-        output = nle(noise, patchsize=11, itr=5)
-        n_levels[n] = output.nlevel
-        n_patches[n] = output.num
+        output = nf.noise_estimation(noise, patchsize=11, itr=5)
+        n_levels[n] = output[0]
+        n_patches[n] = output[2]
 
-    assert np.abs(1 - n_levels / noise_levels) < 0.1
-    assert n_patches > 10000.0
+        assert np.abs(1 - n_levels[n] / noise_levels[n]) < 0.1
+        assert n_patches[n] > 10000.0
 
 
-def weaktexturemask_test():
+def weaktexturemask_test(img):
+
     noise_levels = 5
     noise = img + np.random.standard_normal(img.shape) * noise_levels
-    output = nle(noise, patchsize=11, itr=5)
+    output = nf.noise_estimation(noise, patchsize=11, itr=5)
+
+    assert np.sum(output[3]) / output[3].size < 1.0
 
-    assert np.sum(output.mask) / output.mask.size < 1.0
```

Comment:
`output.mask` is way more readable than `output[3]`. I think this should be done differently. Either as a dictionary where you could say `output['mask']` or something else.

Reply:
I'm going to use a dictionary for the output, agree with you that it's much more readable.